### ISSUE TITLE
tv_device runs into error "free(): double free detected in tcache 2"

### ISSUE THREAD

**Issue by frobones** [pupnp/pupnp#389]
Sample program runs into a double free error. My platform is a Raspbian 11 bullseye. The error I receive is:
```
Initializing UPnP Sdk with
        interface = {NULL} port = 0
UPnP Initialized
        ipaddress = 192.168.1.44 port = 49152
Specifying the webserver root directory -- ./web
Registering the RootDevice
         with desc_doc_url: http://192.168.1.44:49152/tvdevicedesc.xml
free(): double free detected in tcache 2
Aborted
```

Using `valgrind` (https://valgrind.org/) I tracked down the issue to the following line in `webserver.c`: https://github.com/pupnp/pupnp/blob/master/upnp/src/genlib/net/http/webserver.c#L554

It appears that `fp` is closed and then is closed again at the `exit_function` label.

Here is the commit that introduced the double close: https://github.com/pupnp/pupnp/commit/0b1a6c59be3babc3ecaf81debdfc159759872215#diff-0b6cfade43473f4ba934e72d8cf4618e511b14e5aefdc60b5ca7a2a6ee18425f

Either `fp` should be set to `NULL` after the `close` at `line 554` or it should be removed entirely.

**Comment by mrjimenez** [comment]
Great job!

Both solutions you suggest are fine. I think I prefer setting fp to zero right after close, since the close at this position indicates for those reading the code that we no longer need fp.

Would you like to make a patch and a pull request or do you want me to do the commit? Since you've nailed it, I think you should do it. If you choose to do it, please update the `Changelog` file too.

Best regards,
Marcelo.

**Comment by frobones** [comment]
Cool, I'll make a patch and pull request. I'll be sure to get it in this week.

**Comment by nishy2000** [comment]
I think that this issue #389 has been resolved with the merging of PR #390.

**Comment by mrjimenez** [comment]
You are correct. Closing.
